integration/jaxrs/src/main/java/com/blazebit/persistence/integration/jaxrs/EntityViewMessageBodyReader.java

```diff
@@ -169,4 +181,46 @@ private boolean hasMatchingMediaType(MediaType mediaType) {
          */
         return true;
     }
+
+    /**
+     * ParamConverterProvider for default parameter conversion using fromString method if present
+     */
+    private static class FromStringParamConverterProvider implements ParamConverterProvider {
+        @Override
+        public <T> ParamConverter<T> getConverter(Class<T> clazz, Type type, Annotation[] annotations) {
+            try {
+                return new ParamConverter<>(clazz.getMethod("fromString", String.class));
```

Comment:
According to the JAX-RS specification, here is a listing in the Resteasy documentation https://docs.jboss.org/resteasy/docs/3.5.0.Final/userguide/html/StringConverter.html#d4e1480, you should also support primitive types as well as types that have a public constructor that accepts just a string. In addition to that, you should also support a static method `valueOf`.
Apart from that, it would be good to cache the `ParamConverter` in a concurrent hash map within `FromStringParamConverterProvider`